src/main/java/edu/umich/ctools/sectionsUtilityTool/SISPollingThread.java

```diff
@@ -0,0 +1,221 @@
+package edu.umich.ctools.sectionsUtilityTool;
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.apache.http.HttpStatus;
+import org.apache.http.client.methods.HttpGet;
+import org.json.JSONArray;
+import org.json.JSONException;
+import org.json.JSONObject;
+
+import javax.activation.DataHandler;
+import javax.mail.*;
+import javax.mail.internet.InternetAddress;
+import javax.mail.internet.MimeBodyPart;
+import javax.mail.internet.MimeMessage;
+import javax.mail.internet.MimeMultipart;
+import javax.mail.util.ByteArrayDataSource;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Properties;
+
+/**
+ * Created by pushyami on 4/7/17.
+ */
+public class SISPollingThread implements Runnable {
+	private static Log M_log = LogFactory.getLog(SISPollingThread.class);
+
+	volatile boolean flag = true;
+	Properties appExtPropertiesFile = SectionsUtilityToolServlet.appExtPropertiesFile;
+	private int pollingAttempts =Integer.valueOf(appExtPropertiesFile.getProperty(Utils.SIS_POLLING_ATTEMPTS));
+	private int pollingFrequency =Integer.valueOf(appExtPropertiesFile.getProperty(Utils.SIS_POLLING_FREQUENCY));
+
+
+	@Override
+	public void run() {
+		while (flag) {
+			M_log.debug("***************************** Starting polling Thread");
+			List<SISDataHolderForEmail> pollingId = SectionsUtilityToolServlet.canvasPollingIds;
+			listOFPolls(pollingId);
+			Iterator<SISDataHolderForEmail> iterator = pollingId.iterator();
+			while (iterator.hasNext()) {
+				SISDataHolderForEmail emailData = iterator.next();
+				M_log.debug(String.format("Number of attempts by for job %s are %s "
+						, emailData.getPollingId(), emailData.getNumberOfTries()));
+				if (emailData.getNumberOfTries() >= pollingAttempts) {
+					emailData.setSISUploadFailed(true);
```

Comment:
I think I did not document/print enough logs to make it clear. This particular case is `gave up on tracking`. As canvas usually does the SIS process( to size that we are requesting ) reasonably quickly but there are odd ball. So this is addressing that Odd ball case.